src/compiler/checker.ts

```diff
@@ -9230,6 +9263,10 @@ namespace ts {
                         return Ternary.False;
                     }
                 }
+                if (alreadyComparingTypeReferences(source, target)) {
+                    return Ternary.Maybe;

```

Comment:
After discussing it with you: probably yes, as long as we only cache non-constrained type parameters. With the combined maybeKeys, when checking `A<T> ==> B<T>` and then `A<U> ==> B<U>`, when we skip `A<U>===>B<U>` with Ternary.Maybe and later find that they *are* assignable, we'll also mark `A<T>==>B<T>` as assignable. Which is what we want.